rebalance/rebalance.py

```diff
@@ -168,16 +174,43 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi = Non
     description = "%s to %s" % (outgoing_scid, incoming_scid)
     invoice = plugin.rpc.invoice(msatoshi, label, description, retry_for + 60)
     payment_hash = invoice['payment_hash']
-    success_msg = ""
-    try:
-        excludes = []
-        # excude all own channels to prevent unwanted shortcuts [out,mid,in]
-        mychannels = plugin.rpc.listchannels(source=my_node_id)['channels']
-        for channel in mychannels:
-            excludes += [channel['short_channel_id'] + '/0', channel['short_channel_id'] + '/1']
 
+    rpc_result = None
+    count = 0
+    excludes = [my_node_id]  # excude all own channels to prevent shortcuts
+    nodes = {}               # here we store erring node counts
+    routes = {}              # trace all routes to suppress duplicates
+    maxhops = 1              # start with short routes and increase
+    msat_factor = 4          # start with high msatoshi factor to reuce
```

Comment:
Could this be something like this?
```
    maxhops = plugin.starting_maxhops
    msat_factor = plugin.starting_msat_factor
```
And then `plugin.starting_maxhops` and `plugin.starting_msat_factor` could come from config, with a default value of 1 and 4. But it would be possible to set them 5 and 1, to call `getroute` with fixed `maxhops=5` and `msat_factor=1`.

Reply:
Sure, lets test it to find best values. Or maybe other approaches to improve successful path finding

Reply:
In the mailing list there's some chat about it that the (only?) potential drawback is computational time used for getroute.
If that's true then its really a matter of CPU vs Net-Latency tradeoff which in a lot situations will be no problem for the CPU.

Lets optimize values... we need to trace how much CPU getroute overhead is caused vs how much net io is saved ...

Reply:
Maybe we can have a "reference value" about how long an average `sendpay` takes and then let the plugin runtime measure how fast `getroute` is on this machine to turn on or off this optimization ...